internal/server/repository_workerauth.go

```diff
@@ -233,7 +233,23 @@ func StoreNodeInformationTx(ctx context.Context, reader db.Reader, writer db.Wri
 		return errors.Wrap(ctx, err, op)
 	}
 
-	// Store WorkerAuth
+	// Check if we already have a workerAuth record for this key id
+	nodeAuthLookup := allocWorkerAuth()
+	nodeAuthLookup.WorkerKeyIdentifier = node.Id
+	if err := reader.LookupById(ctx, nodeAuthLookup); err != nil {
+		switch {
+		case errors.Convert(err) != nil && errors.Convert(err).Code == errors.RecordNotFound:
```

Comment:
Can we not just use
```suggestion
		case errors.IsNotFoundError(err):
```
?

Reply:
ty!